include/alpaka/mem/view/AccessorAdaptor.hpp

```diff
@@ -0,0 +1,169 @@
+/* Copyright 2022 Andrea Bocci
+
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED “AS IS” AND ISC DISCLAIMS ALL WARRANTIES WITH
+ * REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL ISC BE LIABLE FOR ANY
+ * SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR
+ * IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#pragma once
+
+#include <alpaka/dim/Traits.hpp>
+#include <alpaka/extent/Traits.hpp>
+#include <alpaka/mem/view/ViewAccessor.hpp>
+
+#include <sstream>
+#include <type_traits>
+
+namespace alpaka
+{
+    namespace internal
+    {
+        template<typename TView>
+        struct AccessorAdaptor
+        {
+        private:
+            using value_type = Elem<TView>;
+            using pointer = value_type*;
+            using const_pointer = value_type const*;
+            using reference = value_type&;
+            using const_reference = value_type const&;
+
+        public:
+            AccessorAdaptor()
+            {
+                static_assert(experimental::traits::internal::IsView<TView>::value, "");
```

Comment:
We have C++17 now:
```suggestion
                static_assert(experimental::traits::internal::IsView<TView>::value);
```

Reply:
OK, but just FYI I copied this approach from [include/alpaka/mem/view/ViewAccessor.hpp](https://github.com/alpaka-group/alpaka/blob/develop/include/alpaka/mem/view/ViewAccessor.hpp#L238)

Reply:
Sure, that feature was added before we had C++17. @sliwowitz has some refactorings for such things in the pipeline.

Reply:
Sure, that feature was added before we had C++17. @sliwowitz has some refactorings for such things in the pipeline.